### PR TITLE
techdebt(ParticipantsTab) - refactor participants list

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Preparations to apply RecycleScroller: https://github.com/nextcloud/spreed/issues/5907#issuecomment-1690096604
* Move out all logic from `CurrentParticipants.vue` and delete it:
  * Replace mixin `userStatus` with composable,
  * Move out complex method `sortParticipants` to composable (:warning: Not a clean move, some lines were modified. Please see thoroughly)
  * Move some computed properties and `updateUserStatus` method to the parent component
* Follow-ups:
  * transform `<ParticipantsList />` or it's copy to RecycleSroller
  * replace `<LoadingParticipant />` with `<LoadingPlaceholder />` #10489

### 🖼️ Screenshots
No visual changes

### 🚧 Tasks

- [ ] Code review
- [ ] Manual testing

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is noxt required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/composables/useSortParticipants.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+/**
+ * Handler for participants lists sorting
+ */
+export function useSortParticipants() {
+	const store = useStore()
+	const { isDoNotDisturb } = useUserStatus()
+
+	const selfIsModerator = computed(() => {
+		const participantType = store.getters.conversation(store.getters.getToken())?.participantType
+			|| (store.getters.getUserId() !== null ? PARTICIPANT.TYPE.USER : PARTICIPANT.TYPE.GUEST)
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/composables/useSortParticipants.js:38]: There is no need to do it for ParticipantsTab.
But it's a reusable code, so we could implement it somewhere else (for example, "Create a new group conversation"). 

So we can either keep it as a fallback, like in `<VideoBottomBar/>` or remove

File: src/components/RightSidebar/Participants/ParticipantsTab.vue [hunk 2]
```diff
@@ -92,6 +114,23 @@ export default {
...
 
 	computed: {
+		participants() {
+			return this.$store.getters.participantsList(this.token).sort(this.sortParticipants)
+		},
+
+		filteredParticipants() {
+			if (!this.isSearching) {
+				return this.participants
+			}
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/RightSidebar/Participants/ParticipantsTab.vue:124]: `ifSearching` depends on `searchText`, so will be recomputed in any case.
Make a small sense for this fallback here

File: src/composables/useUserStatus.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+	 */
+	function isDoNotDisturb(userData) {
+		return userData.status === 'dnd'
+	}
+
+	return {
+		getStatusMessage,
+		isDoNotDisturb,
+	}
+}
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/composables/useUserStatus.js:67]: Agree. Maybe move it to the utils in that case?

**Review by ShGKme — APPROVED** [review]

**Comment by Antreesy** [comment]
Squashed and rebased onto master, no changes

**Pull request merged**